File: android/app/src/main/java/com/synergyboat/flashcardAi/core/benchmark/DbSizeLogger.kt
package com.synergyboat.flashcardAi.core.benchmark

import java.util.logging.Logger
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class DbSizeLogger @Inject constructor(
    private val logger: Logger
) {

    fun logDbRowSize(
        row: Map<String, Any?>,
        name: String = "",
        tag: String = "db_row_size",
        log: Boolean = true
    ) {
        val sizeInBytes = RowSizeBenchmark.getRowSizeInBytes(row)
        val sizeInKB = RowSizeBenchmark.getRowSizeInKB(row)

        if (log) {
            logger.info("$tag | Row size for $name: $sizeInBytes bytes (${String.format("%.2f", sizeInKB)} KB)")
        }
    }

    fun logTotalDbRowSize(
        rows: List<Map<String, Any?>>,
        name: String = "",
        tag: String = "db_row_size",
        log: Boolean = true
    ) {
        val totalSizeInBytes = rows.sumOf { RowSizeBenchmark.getRowSizeInBytes(it) }
        val totalSizeInKB = totalSizeInBytes / 1024.0

        if (log) {
            logger.info("$tag | Total row size for $name: $totalSizeInBytes bytes (${String.format("%.2f", totalSizeInKB)} KB)")
        }
    }
}

File: ios/FlashcardAI/Core/Benchmark/DbSizeLogger.swift
import Foundation

final class DbSizeLogger {
    static func logDbRowSize(_ row: [String: Any], name: String = "", tag: String = "db_row_size", log: Bool = true) {
        let sizeBytes = RowSizeBenchmark.getRowSizeInBytes(row: row)
        let sizeKB = RowSizeBenchmark.getRowSizeInKB(row: row)

        if log {
            print("\(tag) | Row size for \(name): \(sizeBytes) bytes (\(String(format: "%.2f", sizeKB)) KB)")
        }
    }

    static func logTotalDbRowSize(_ rows: [[String: Any]], name: String = "", tag: String = "db_row_size", log: Bool = true) {
        let totalBytes = rows.reduce(0) { $0 + RowSizeBenchmark.getRowSizeInBytes(row: $1) }
        let totalKB = Double(totalBytes) / 1024.0

        if log {
            print("\(tag) | Total row size for \(name): \(totalBytes) bytes (\(String(format: "%.2f", totalKB)) KB)")
        }
    }
}


